models/intercom__conversation_enhanced.sql

```diff
@@ -63,25 +63,43 @@ team as (
 ),
 {% endif %}
 
+latest_conversation_enriched as (
```

Comment:
Is there a need to add all this additional code? If the need is to persist the passthrough columns only, could we in theory just do the following:

```sql
enriched as ( 
    select
        latest_conversation.conversation_id,
        latest_conversation.created_at as conversation_created_at,
        latest_conversation.updated_at as conversation_last_updated_at,
        conversation_part_events.first_close_at,
        conversation_part_events.last_close_at,
        latest_conversation.source_type as conversation_type,
        latest_conversation.source_delivered_as as conversation_initiated_type,

        -- All the stuff in between these lines

        latest_conversation.conversation_rating_value as conversation_rating,
        latest_conversation.conversation_rating_remark as conversation_remark

       {{ fivetran_utils.fill_pass_through_columns('intercom__conversation_history_pass_through_columns') }}

    from latest_conversation
```

This way we don't have to risk adjusting a large amount of code in the model and are able to persist the fields that the customer defines from the variable.

Reply:
I thought to do it that way, but then I didn't see how to bring in the CTE name (`latest_conversation` in this case) without adding like a jinja loop. I had tried using `persist_pass_through_columns` since you can pass the CTE name, but it only worked for variables using the mapped name values, and I didn't want to break existing installations. Open to other ideas though of course!

Reply:
Great callout and relevant issue that we will want to avoid when allowing customers to passthrough variables into these downstream models. That is also an interesting call out that the `persist_pass_through_columns` macro doesn't work with the non-mapping version. I would have assumed this could be handled for both cases.

I wonder if this is a situation where we should update that macro to check if the variables [is mapping](https://github.com/fivetran/dbt_fivetran_utils/blob/cc2329a92fdaee11ea59fd8a7ce69e8b8fe8fe6c/macros/fill_pass_through_columns.sql#L5-L9) or not.

Reply:
Let's discuss during standup

Reply:
Following our conversation during standup. We will keep the current code changes you are suggesting in this PR and will create an accompanying feature request on fivetran_utils to add the is mapping logic to the persist pass through maco.